apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/Trace/SimpleManagedTracerTest.cs

```diff
@@ -254,6 +256,203 @@ public void MultipleSpans()
         }
 
         [Fact]
+        public async Task MultipleSpans_Async()
+        {
+            var mockConsumer = new Mock<IConsumer<TraceProto>>();
+            var tracer = SimpleManagedTracer.Create(mockConsumer.Object, ProjectId, TraceId);
+
+            Predicate<IEnumerable<TraceProto>> matcher = (IEnumerable<TraceProto> t) =>
+            {
+                var spans = t.Single().Spans.OrderBy(s => s.Name).ToList();
+                var rootId = spans[4].SpanId;
+                var childOneId = spans[0].SpanId;
+                var childTwoId = spans[1].SpanId;
+                return spans.Count == 5 &&
+                    IsValidSpan(spans[0], "child-one", parentId: rootId) &&
+                    IsValidSpan(spans[1], "child-two", parentId: rootId) &&
+                    IsValidSpan(spans[2], "grandchild-one", parentId: childOneId) &&
+                    IsValidSpan(spans[3], "grandchild-two", parentId: childTwoId) &&
+                    IsValidSpan(spans[4], "root");
+            };
+            mockConsumer.Setup(c => c.Receive(Match.Create(matcher)));
+
+            Func<string, string, Task> op = async (childName, grandchildName) =>
+            {
+                using (tracer.StartSpan(childName))
+                {
+                    await Task.Delay(30);
```

Comment:
Are these just to enforce it not running synchronously? If so, `Task.Yield` might be a better option.